setuptools_rust/setuptools_ext.py

```diff
@@ -1,8 +1,8 @@
 import os
 import subprocess
 import sysconfig
-from distutils import log
-from distutils.command.clean import clean
+import logging
+from setuptools.command.clean import clean
```

Comment:
This is only used for type checking, so I think you could leave it as distutils and put it under an `if TYPE_CHECKING` guard?

Reply:
It seems to be used not only for typing as it breaks the CI, trying to use vendorized version of `distutils` in the meantime.